### DISCUSSION TITLE
OMI glTF Working Group Meeting 2023-03-30

### DISCUSSION THREAD

**Discussion by madjin** [omigroup/gltf-extensions#153] [General]
This meeting is on 3/30/2023 at 9:30 PM (UTC) / 2:30 PM (PST) in the [OMI Discord](https://discord.gg/NJtT9grz5E) within the #omi-weekly-meeting channel. During the meeting, we will be using the #omi-gltf-extensions channel to manage a speaker queue, post links, and for any sidebar discussions.

To be notified of this meeting and others, subscribe to the [OMI Meetings and Events Calendar](https://calendar.google.com/calendar/u/1?cid=Y18wZHB1Z2Y5ZjgzZXE0cWVrbWI2b21xYmptZ0Bncm91cC5jYWxlbmRhci5nb29nbGUuY29t) or add yourself to the @omi-gltf-subgroup role in the #roles channel of the OMI Discord.

We have a bit of a problem due to DST, since the meeting time changes differently in different parts of the world

## Agenda:
* Discuss booth presentation?
* [MSFT_CollisionPrimitive & MSFT_RigidBodies](https://github.com/KhronosGroup/glTF/pull/2257)
* [OMI Physics extensions](https://github.com/KhronosGroup/glTF/pull/2258)
* [OMI_seat proposal](https://github.com/omigroup/gltf-extensions/pull/143)

## Backlog:
* [OMI_personality Stage 1 Vote (delayed until next week)](https://github.com/omigroup/gltf-extensions/pull/133)
* We discussed https://openmv.org (Open Metaverse Foundation) and how we can synergize with their efforts. Should we make a showcase of our projects to open up a broader discussion?
* jin - proposals how to make our specs and work more accessible to others. showcase gltf models
* Review pull requests https://github.com/omigroup/gltf-extensions/pulls
* [EXT_skin_humanoid](https://github.com/takahirox/EXT_skin_humanoid)
* [OMI_link](https://github.com/omigroup/gltf-extensions/tree/main/extensions/2.0/OMI_link)
* [OMI_spawn_point](https://github.com/omigroup/gltf-extensions/tree/main/extensions/2.0/OMI_spawn_point)
* Review [MSFT_physics](https://github.com/eoineoineoin/glTF_Physics)
  * [glTF_Physics_Blender_Exporter](https://github.com/eoineoineoin/glTF_Physics_Blender_Exporter)
  * [glTF_Physics_Godot_Importer](https://github.com/eoineoineoin/glTF_Physics_Godot_Importer)
* Licensing Extension https://twitter.com/superhoge/status/1596757861882740736
* [KHR_audio / Khronos feedback](https://github.com/KhronosGroup/glTF/pull/2137)
  * [gltf-sample-assets](https://github.com/KhronosGroup/glTF-Sample-Models/pull/360)
* [Third Room Extensions](https://github.com/matrix-org/thirdroom/pull/469)
* [glXF](https://github.com/KhronosGroup/glXF/blob/main/specification/2.0/README.md)
* Show and tell
* Create next weeks agenda

To propose another item for the agenda, comment below (preferably before the meeting).

**Comment by madjin** [comment]
Hyperfy just added vehicles, we have a good interoperability lab setup now for prototyping extensions with
![image](https://user-images.githubusercontent.com/32600939/228975557-b660caac-fd32-4a5d-b35d-277335131913.png)
Source: https://twitter.com/philburrrrt/status/1641559637995470848

## Physics Materials and Vehicle Physics
- https://gameworksdocs.nvidia.com/PhysX/4.1/documentation/physxguide/Manual/Vehicles.html
- https://docs.unrealengine.com/4.26/en-US/InteractiveExperiences/Physics/ChaosPhysics/ChaosVehicles/HowToSetupChaosVehicles/
- https://pybullet.org/Bullet/BulletFull/classbtRaycastVehicle.html

![image](https://user-images.githubusercontent.com/32600939/228975404-c84a25d2-2021-41fb-af05-49457fca6710.png)
Source: https://twitter.com/SubstrataVr/status/1628728604279586818

**Reply by robertlong** [reply]
Also Godot's Vehicle Physics: https://docs.godotengine.org/en/stable/classes/class_vehiclebody3d.html
And Rapier is supposed to be adding vehicles soon: https://dimforge.com/blog/2023/01/22/the-year-2022-in-dimforge#dynamic-vehicle-controller

**Comment by madjin** [comment]
- https://github.com/jamschutz/Unity-Standard-Assets/blob/master/Standard%20Assets/Utility/FollowTarget.cs
- https://github.com/search?q=repo%3AhfcRed%2FAmong-Us-Follower+target&type=code
- https://github.com/hfcRed/Among-Us-Follower
- https://github.com/search?q=repo%3AhfcRed%2FAmong-Us-Follower+target&type=code

![image](https://user-images.githubusercontent.com/32600939/228979348-0510084c-0418-42b1-a1c9-c64a2d80c7f6.png)

Simple pet system in vrchat that doesn't need any special udon scripting, just follow target and look at target
Would be a fun case study for physics perhaps, the slight delay / naturalness of pet follow distance reminded me about joints / constraints
